make run not working?

Hi, I've been following the instructions as per the readme.me but when I run `make run`, I get the following error:
make run
python -m timeeval_gui
Traceback (most recent call last):
  File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lcs_crr/PycharmProjects/TimeEval-GUI/timeeval_gui/__main__.py", line 4, in <module>
    from streamlit import cli as stcli
  File "/home/lcs_crr/PycharmProjects/TimeEval-GUI/.venv/lib/python3.9/site-packages/streamlit/__init__.py", line 70, in <module>
    from streamlit.delta_generator import DeltaGenerator as _DeltaGenerator
  File "/home/lcs_crr/PycharmProjects/TimeEval-GUI/.venv/lib/python3.9/site-packages/streamlit/delta_generator.py", line 90, in <module>
    from streamlit.elements.arrow_altair import ArrowAltairMixin
  File "/home/lcs_crr/PycharmProjects/TimeEval-GUI/.venv/lib/python3.9/site-packages/streamlit/elements/arrow_altair.py", line 27, in <module>
    from altair.vegalite.v4.api import Chart
ModuleNotFoundError: No module named 'altair.vegalite.v4'
make: *** [Makefile:5: run] Error 1

Perhaps there is an issue with the dependency versions? Here's my `pip list`:
Package                   Version
------------------------- -----------
altair                    5.5.0
asyncssh                  2.5.0
attrs                     25.1.0
blinker                   1.9.0
cachetools                5.5.2
certifi                   2025.1.31
cffi                      1.17.1
charset-normalizer        3.4.1
click                     8.0.2
cloudpickle               3.1.1
cryptography              44.0.2
cycler                    0.12.1
dask                      2021.5.1
distributed               2021.5.1
docker                    4.4.4
durations                 0.3.3
fonttools                 4.56.0
fsspec                    2025.2.0
gitdb                     4.0.12
GitPython                 3.1.44
idna                      3.10
importlib_metadata        8.6.1
Jinja2                    3.0.3
joblib                    1.0.1
jsonschema                4.23.0
jsonschema-specifications 2024.10.1
kiwisolver                1.4.7
locket                    1.0.0
markdown-it-py            3.0.0
MarkupSafe                3.0.2
matplotlib                3.5.3
mdurl                     0.1.2
msgpack                   1.1.0
narwhals                  1.29.0
neurokit2                 0.1.2
numpy                     1.21.6
numpyencoder              0.3.0
packaging                 24.2
pandas                    1.3.5
partd                     1.4.2
patsy                     1.0.1
pillow                    11.1.0
pip                       23.2.1
plotly                    5.10.0
protobuf                  3.20.3
prts                      1.0.0.3
psutil                    7.0.0
pyarrow                   19.0.1
pycparser                 2.22
pydeck                    0.9.1
Pygments                  2.19.1
Pympler                   1.1
pyparsing                 3.2.1
python-dateutil           2.9.0.post0
pytz                      2025.1
PyYAML                    5.3.1
referencing               0.36.2
requests                  2.32.3
rich                      13.9.4
rpds-py                   0.23.1
scikit-learn              1.0.2
scipy                     1.11.4
semver                    3.0.4
setuptools                68.2.0
six                       1.17.0
smmap                     5.0.2
sortedcontainers          2.4.0
statsmodels               0.12.2
streamlit                 1.11.1
tblib                     3.0.0
tenacity                  9.0.0
threadpoolctl             3.5.0
TimeEval                  1.2.4
timeeval-gutenTAG         0.2.0
toml                      0.10.2
toolz                     1.0.0
tornado                   6.4.2
tqdm                      4.54.1
typing_extensions         4.12.2
tzlocal                   5.3
urllib3                   2.3.0
validators                0.34.0
watchdog                  2.1.9
websocket-client          1.8.0
wheel                     0.41.2
zict                      3.0.0
zipp                      3.21.0

--- a/README.md
+++ b/README.md
@@ -31,7 +31,7 @@ Those repositories also include their respective Python APIs:
 [![GutenTAG Badge](https://img.shields.io/badge/Repository-GutenTAG-blue?style=for-the-badge)](https://github.com/TimeEval/gutentag)
 [![Eval Badge](https://img.shields.io/badge/Repository-Eval-blue?style=for-the-badge)](https://github.com/TimeEval/timeeval)
 
-As initial resources for evaluations, we provide over 1,000 benchmark datasets and an increasing number of time series anomaly detection algorithms (over 70): 
+As initial resources for evaluations, we provide over 1,000 benchmark datasets and an increasing number of time series anomaly detection algorithms (over 70):
 
 [![Datasets Badge](https://img.shields.io/badge/Repository-Datasets-3a4750?style=for-the-badge)](https://timeeval.github.io/evaluation-paper/notebooks/Datasets.html)
 [![Algorithms Badge](https://img.shields.io/badge/Repository-Algorithms-3a4750?style=for-the-badge)](https://github.com/TimeEval/TimeEval-algorithms)
@@ -39,12 +39,21 @@ As initial resources for evaluations, we provide over 1,000 benchmark datasets a
 ## Installation and Usage (tl;dr)
 
 TimeEval is tested on Linux and Mac operating systems and supports Python 3.7 until 3.9.
-
 We don't support Python 3.10 or higher at the moment because downstream libraries are incompatible.
 
 > We haven't tested if TimeEval runs on Windows.
 > If you use Windows, please help us and test if TimeEval runs correctly.
 > If there are any issues, don't hesitate to contact us.
+
+By default, TimeEval does not automatically download all available algorithms (Docker images), because there are just too many.
+However, you can download them easily [from our registry](https://github.com/orgs/TimeEval/packages?repo_name=TimeEval-algorithms) using docker.
+Please download the correct tag for the algorithm, compatible with your version of TimeEval:
+
+```bash
+docker pull ghcr.io/timeeval/kmeans:0.3.0
+```
+
+After you have downloaded the algorithm images, you need to restart the GUI, so that it can find the new images.
 
 ### Web frontend
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -9,3 +9,5 @@ requests
 protobuf>=3.20,<4
 watchdog==2.1.9
 plotly==5.10.*
+altair==4.2.2  # newer versions of altair are not compatible with streamlit 1.11.1
+requests==2.31.0  # newer versions break the docker connection with "Error while fetching server API version: Not supported URL scheme http+docker"

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 streamlit==1.11.1
-timeeval==1.2.4
+timeeval>=1.4,<1.5
 timeeval-gutentag==0.2.0
 pyyaml
 numpy

--- a/timeeval_gui/_pages/eval/__init__.py
+++ b/timeeval_gui/_pages/eval/__init__.py
@@ -1,18 +1,16 @@
 import logging
 import sys
-from typing import List, Dict, Any, Optional, Union
+from typing import List, Union
 
 import docker
-import numpy as np
 import psutil
 import streamlit as st
 from docker.errors import DockerException
 from durations import Duration
-from streamlit.state import NoValue
-from timeeval import Algorithm, ResourceConstraints, Metric, TimeEval
+from timeeval import Algorithm, ResourceConstraints, DefaultMetrics, TimeEval
 from timeeval.params import FixedParameters, FullParameterGrid, IndependentParameterGrid
 from timeeval.resource_constraints import GB
-from timeeval_experiments import algorithms as timeeval_algorithms
+from timeeval import algorithms as timeeval_algorithms
 
 import timeeval_gui.st_redirect as rd
 from .param_opt import InputParam
@@ -21,7 +19,7 @@ from ...config import SKIP_DOCKER_PULL
 from ...files import Files
 
 # keep this import!
-from timeeval_experiments.algorithms import *
+from timeeval.algorithms import *
 
 
 def create_algorithm_list() -> List[Algorithm]:
@@ -40,7 +38,7 @@ def create_algorithm_list() -> List[Algorithm]:
 
         algorithms = [a for a in algorithms if image_exists(a.main.image_name, a.main.tag)]
         del docker_client
-        return algorithms
+    return algorithms
 
 
 algos: List[Algorithm] = create_algorithm_list()
@@ -145,10 +143,12 @@ class EvalPage(Page):
         st.write("## General Settings")
 
         repetitions = st.slider("Repetitions", value=1, min_value=1, max_value=1000, step=1)
-        metrics = st.multiselect("Metrics",
-                                 options=[m for m in Metric if m not in {Metric.RANGE_F1, Metric.RANGE_RECALL, Metric.RANGE_PRECISION}],
-                                 default=Metric.default_list(),
-                                 format_func=lambda m: m.name)
+        metric_options = [
+            "ROC_AUC", "PR_AUC", "RANGE_PR_AUC", "AVERAGE_PRECISION", "RANGE_PRECISION",
+            "RANGE_RECALL", "RANGE_F1", "FIXED_RANGE_PR_AUC",
+        ]
+        metric_names = st.multiselect("Metrics", options=metric_options, default="ROC_AUC")
+        metrics = [getattr(DefaultMetrics, m) for m in metric_names]
         force_training_type_match = st.checkbox("Force training type match between algorithm and dataset", value=False)
         force_dimensionality_match = st.checkbox(
             "Force dimensionality match between algorithm and dataset (uni- or multivariate)",

--- a/timeeval_gui/_pages/results.py
+++ b/timeeval_gui/_pages/results.py
@@ -9,14 +9,13 @@ from timeeval import DatasetManager, Datasets
 import plotly.graph_objects as go
 
 from .page import Page
-from ..config import TIMEEVAL_FILES_PATH
 from ..files import Files
 
 
 @st.cache(show_spinner=True, max_entries=1)
 def load_results(results_path: Path) -> pd.DataFrame:
     res = pd.read_csv(results_path / "results.csv")
-    res["dataset_name"] = res["dataset"].str.split(".").str[0]
+    res["dataset_name"] = res["dataset"]
     res["overall_time"] = res["execute_main_time"].fillna(0) + res["train_main_time"].fillna(0)
     res["algorithm-index"] = res.algorithm + "-" + res.index.astype(str)
     res = res.drop_duplicates()
@@ -30,7 +29,9 @@ def create_dmgr(data_path: Path) -> Datasets:
 
 @st.cache(show_spinner=True, max_entries=100, hash_funcs={pd.DataFrame: pd.util.hash_pandas_object, "builtins.function": lambda _: None})
 def plot_boxplot(df, n_show: Optional[int] = None, title="Box plots", ax_label="values", metric="ROC_AUC", _fmt_label=lambda x: x, log: bool = False) -> go.Figure:
-    df_asl = df.pivot(index="algorithm-index", columns="dataset_name", values=metric)
+    df_asl = df.copy()
+    df_asl["dataset_name"] = df_asl["dataset_name"].str.split(".").str[0]
+    df_asl = df_asl.pivot(index="algorithm-index", columns="dataset_name", values=metric)
     df_asl = df_asl.dropna(axis=0, how="all").dropna(axis=1, how="all")
     df_asl["median"] = df_asl.median(axis=1)
     df_asl = df_asl.sort_values(by="median", ascending=True)
@@ -102,7 +103,6 @@ def plot_scores(algorithm_name, collection_name, dataset_name, df, dmgr, result_
             continue
 
         # load scores
-        training_type = df.loc[df["algorithm"] == algo, "algo_training_type"].values[0].lower().replace("_", "-")
         try:
             df_scores[algo] = load_scores_df(algo, dataset_id, df, result_path).iloc[:, 0]
         except (ValueError, FileNotFoundError):
@@ -111,7 +111,7 @@ def plot_scores(algorithm_name, collection_name, dataset_name, df, dmgr, result_
             skip_algos.append(algo)
     algorithms = [a for a in algos if a not in skip_algos]
 
-    fig = plot_scores_plotly(algorithms, auroc, df_scores, df_dataset, dataset_dim, dataset_name)
+    fig = plot_scores_plotly(algorithms, auroc, df_scores, df_dataset, dataset_dim, dataset_name.split(".")[0])
     st.plotly_chart(fig)
 
 
@@ -175,7 +175,7 @@ class ResultsPage(Page):
         with col1:
             collection = st.selectbox("Collection", options=res["collection"].unique())
         with col2:
-            dataset = st.selectbox("Dataset", res[res.collection == collection]["dataset_name"].unique())
+            dataset = st.selectbox("Dataset", res[res.collection == collection]["dataset_name"].unique(), format_func=lambda x: x.split(".")[0])
         with col3:
             options = res[(res.collection == collection) & (res.dataset_name == dataset) & (res.status.isin(["Status.OK", "OK"]))]["algorithm"].unique()
             options = [None] + list(options)

--- a/timeeval_gui/🏠_Home.py
+++ b/timeeval_gui/🏠_Home.py
@@ -4,14 +4,14 @@ import streamlit as st
 def main():
     st.markdown("""
     # Welcome to the TimeEval GUI
-    
+
     TimeEval includes an extensive data generator and supports both interactive and batch evaluation scenarios.
     This novel toolkit, aims to ease the evaluation effort and help the community to provide more meaningful evaluations
     in the Time Series Anomaly Detection field.
-    
+
     This Tool has 3 main components:
-    
-    1. [GutenTAG](/GutenTAG) to generate time series 
+
+    1. [GutenTAG](/GutenTAG) to generate time series
     2. [Eval](/Eval) to run multiple anomaly detectors on multiple datasets
     3. [Results](/Results) to compare the quality of multiple anomaly detectors
     """)

